x-pack/plugins/security_solution/public/management/pages/event_filter/view/components/form/index.tsx

```diff
@@ -0,0 +1,194 @@
+import { ExceptionListItemSchema, ExceptionBuilder } from '../../../../../../shared_imports';
+
+import { useEventFilterSelector } from '../../hooks';
+import { getFormEntry } from '../../../store/selector';
+import {
+  FORM_DESCRIPTION,
+  NAME_LABEL,
+  NAME_ERROR,
+  NAME_PLACEHOLDER,
+  OS_LABEL,
+} from './translations';
+import { OS_TITLES } from '../../../../../common/translations';
+import { EVENT_FILTER_LIST_ID, EVENT_FILTER_LIST_TYPE } from '../../../constants';
+
+const OPERATING_SYSTEMS: readonly OperatingSystem[] = [
+  OperatingSystem.MAC,
+  OperatingSystem.WINDOWS,
+  OperatingSystem.LINUX,
+];
+
+interface EventFilterFormProps {
+  allowSelectOs?: boolean;
+}
+export const EventFilterForm: React.FC<EventFilterFormProps> = memo(({ allowSelectOs = false }) => {
+  const { http, data } = useKibana().services;
+  const dispatch = useDispatch<Dispatch<AppAction>>();
+  const exception = useEventFilterSelector(getFormEntry);
+
+  const [isIndexPatternLoading, { indexPatterns }] = useFetchIndex(['logs-endpoint.events.*']);
+
+  const osOptions: Array<EuiSuperSelectOption<OperatingSystem>> = useMemo(
+    () => OPERATING_SYSTEMS.map((os) => ({ value: os, inputDisplay: OS_TITLES[os] })),
+    []
+  );
+
+  const [hasNameError, setHasNameError] = useState(!exception || !exception.name);
+  const [comment, setComment] = useState<string>('');
+
+  const handleOnBuilderChange = useCallback(
+    (arg: ExceptionBuilder.OnChangeProps) => {
+      if (isEmpty(arg.exceptionItems)) return;
+      dispatch({
+        type: 'eventFilterChangeForm',
+        payload: { entry: arg.exceptionItems[0], hasItemsError: arg.errorExists },
+      });
+    },
+    [dispatch]
+  );
+
+  const handleOnChangeName = useCallback(
+    (e: React.ChangeEvent<HTMLInputElement>) => {
+      if (!exception) return;
+      setHasNameError(!e.target.value);
+      dispatch({
+        type: 'eventFilterChangeForm',
+        payload: {
+          entry: { ...exception, name: e.target.value.toString() },
+          hasNameError: !e.target.value,
+        },
+      });
+    },
+    [dispatch, exception]
+  );
+
+  const handleOnChangeComment = useCallback(
+    (value: string) => {
+      setComment(value);
+      if (!exception) return;
+      dispatch({
+        type: 'eventFilterChangeForm',
+        payload: {
+          entry: { ...exception, comments: [{ comment: value }] },
+        },
+      });
+    },
+    [dispatch, exception, setComment]
+  );
+
+  const exceptionBuilderComponentMemo = useMemo(
+    () => (
+      <ExceptionBuilder.ExceptionBuilderComponent
+        allowLargeValueLists
+        httpService={http}
+        autocompleteService={data.autocomplete}
+        exceptionListItems={[exception as ExceptionListItemSchema]}
+        listType={EVENT_FILTER_LIST_TYPE}
+        listId={EVENT_FILTER_LIST_ID}
+        listNamespaceType={'agnostic'}
+        ruleName={'Endpoint Event Filtering'}
```

Comment:
Does this need to be `i18n`?

Reply:
Right! This is not displayed but i will use i18n anyway.